### PR TITLE
Fixes #14914: Mouse wheel for hud

### LINKED ISSUE #14914: Mouse wheel not working in hud
i think thats a regression

### PR DESCRIPTION
**Pull request by deepakchethan**

### CODE REVIEW

File: libr/cons/dietline.c [hunk 1]
```diff
@@ -1255,6 +1256,12 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 		__print_prompt ();
 	}
 	r_cons_break_push (NULL, NULL);
+	int mouse_status = cons->mouse;
+	if (I.hud) {
+		r_cons_enable_mouse (true);
```
Comment (radare) [inline libr/cons/dietline.c]: what about r_cons_enable-mouse (I.hid); instead ?

File: libr/cons/dietline.c [hunk 2]
```diff
@@ -1615,6 +1619,29 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
...
+						} else if (key == '7') {	 // down
+							if (I.hud) {
+								if (I.hud->top_entry_n >= 0) {
+									I.hud->top_entry_n++;
+								}
+							}
+						} else {
+							ch = 0;
+						}
+						int ch2;
```
Comment (radare) [inline libr/cons/dietline.c]: no need to allocate more variables. just while (r_cons_readchar () != 'M') {}

File: libr/cons/dietline.c [hunk 3]
```diff
@@ -1615,6 +1619,29 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
...
+						key = r_cons_readchar ();
+						cons->mouse_event = 1;
+						if (key == '6') {	// up
+							if (I.hud) {
+								if (I.hud->top_entry_n + 1 < I.hud->current_entry_n) {
+									I.hud->top_entry_n--;
+								}
+							}
+						} else if (key == '7') {	 // down
+							if (I.hud) {
```
Comment (radare) [inline libr/cons/dietline.c]: chain those nested contidionals: if (I.hud && I.hud->..

File: libr/cons/dietline.c [hunk 4]
```diff
@@ -1255,6 +1256,12 @@ R_API const char *r_line_readline_cb(RLineReadCallback cb, void *user) {
 		__print_prompt ();
 	}
 	r_cons_break_push (NULL, NULL);
+	int mouse_status = cons->mouse;
```
Comment (radare) [inline libr/cons/dietline.c]: this should be bool

File: libr/cons/dietline.c (hunk 1 above)
Comment (deepakchethan) [reply libr/cons/dietline.c]: Yes, it is better. Changed it.

**Pull request merged**